src/globals.cpp

```diff
@@ -276,6 +276,8 @@ CThinBlockData thindata; // Singleton class
 
 uint256 bitcoinCashForkBlockHash = uint256S("000000000000000000651ef99cb9fcbe0dadde1d424bd9f15ff20136191a5eec");
 
+map<int64_t, CMiningCandidate> miningCandidatesMap;
+
```

Comment:
+map<int64_t, CMiningCandidate> miningCandidatesMap GUARDED_BY(cs_main);

Reply:
Did not know GUARDED_BY syntax. 
Can explicit locks be removed? 
Such as:
RmOldMiningCandidates()
{
    LOCK(cs_main);
...
it = miningCandidatesMap.erase(it);
...